src/views/Simulation/utils/pixiUtils.js

```diff
@@ -1,14 +1,102 @@
+import { Application, Graphics, GraphicsContext, Container, Text } from 'pixi.js';
 import 'pixi.js/unsafe-eval';
 import { SceneContainer } from './SceneContainer';
+import { PACKAGE_ICON_LINES, GEAR_ICON_LINES, FACTORY_ICON_LINES } from './shapes';
 
-const createGraphicsContext = (options) => {
-  return new GraphicsContext()
-    .rect(0, 0, 8, 8)
+const GRAY_LINE_COLOR = 0xb9bac0;
+const DEFAULT_TEXT_STYLE = { fontFamily: 'Arial', fontSize: 12, fill: 0xffffff, align: 'center' };
+
+const createLabel = (value) => new Text({ text: value, style: DEFAULT_TEXT_STYLE, resolution: 2 });
+
+const drawIcon = (graphicsContext, lines, xOffset, yOffset, width, height) => {
+  lines.forEach((line) => {
+    line.forEach((point, index) => {
+      const x = xOffset + point[0] * width;
+      const y = yOffset + point[1] * height;
+      if (index === 0) graphicsContext.moveTo(x, y);
+      else graphicsContext.lineTo(x, y);
+    });
+  });
+};
+
+const createStockGraphicsContext = (options) => {
+  const graphicsContext = new GraphicsContext()
+    .roundRect(0, 0, 48, 48, 8)
     .fill(options.fillColor)
-    .stroke({ color: options.lineColor, width: 0.75 });
+    .stroke({ pixelLine: true, color: options.borderColor, width: 1.5 });
+  drawIcon(graphicsContext, PACKAGE_ICON_LINES, 15, 14, 18, 20);
+  graphicsContext.stroke({ pixelLine: true, color: options.iconColor, width: 1.5 });
+  return graphicsContext;
+};
+
+const createProductionOperationGraphicsContext = (options) => {
+  const graphicsContext = new GraphicsContext()
+    .roundRect(0, 0, 48, 48, 8)
+    .fill(options.fillColor)
+    .stroke({ pixelLine: true, color: options.borderColor, width: 1.5 });
+  drawIcon(graphicsContext, GEAR_ICON_LINES, 15, 14, 18, 20);
+  graphicsContext
+    .stroke({ pixelLine: true, color: options.iconColor, width: 1.5 })
+    .circle(24, 24, 4)
+    .stroke({ pixelLine: true, color: options.iconColor, width: 1.5 });
+  return graphicsContext;
+};
+
+const createProductionResourceBorderGraphicsContext = (options) => {
+  const radius = 8;
+  const border = new GraphicsContext()
+    .roundRect(0, 0, 80, 80, radius)
+    .fill(options.fillColor)
+    .moveTo(0, 80)
+    .lineTo(0, 20)
+    .arcTo(0, 0, 20, 0, radius)
+    .lineTo(20, 0)
+    .moveTo(60, 0)
+    .arcTo(80, 0, 80, 80, radius)
+    .lineTo(80, 80)
+    .stroke({ pixelLine: true, color: options?.borderColor ?? GRAY_LINE_COLOR, width: 1.5 });
+  return border;
+};
+
+const createFactoryIconBorderGraphicsContext = () => {
+  const graphicsContext = new GraphicsContext();
+  drawIcon(graphicsContext, FACTORY_ICON_LINES, 0, 0, 20, 20);
+  graphicsContext.stroke({ pixelLine: true, color: GRAY_LINE_COLOR, width: 1.5 });
+  return graphicsContext;
+};
+
+const createStockContainer = (graphicsContexts, name, color = 0x000000) => {
+  const container = new Container();
+  const stock = new Graphics(graphicsContexts.stock);
+  const label = createLabel(name);
+  label.anchor.set(0.5);
```

Comment:
In my memory, sprites positions are commonly set using the top left corner. I don't know if it a good practice to systematically set the anchor to the middle of th sprite.